crates/km/src/time.rs

```diff
@@ -0,0 +1,29 @@
+use crate::mode::ProcessorMode;
+use core::time::Duration;
+use km_sys::{KeDelayExecutionThread, LARGE_INTEGER};
+
+/// Sleep in kernel-mode, non-alertable.
+///
+/// > Where possible, Alertable should be set to FALSE and WaitMode should be set to KernelMode, in
+/// > order to reduce driver complexity. The principal exception to this guideline is when the wait
+/// > is a long-term wait.
+pub fn sleep_km(d: Duration) {
```

Comment:
Should this receive an enum like:
``` 
enum Sleep {
     Relative(Duration),
     Absolute(Duration)
}
```

Reply:
IMO it would be a seprate function (similar to tokio's `sleep_until`) for that, same with Kernel/Usermode and alertable/non-alertable. Only implementing the basic case as that's all we need ATM.